File: Arrays/myOwnArray/myOwnArray.py
class Array:
    # define constructor
    def __init__(self):
        self.length = 0
        self.data = {}

    #  convert to string
    def __str__(self):
        return str(self.__dict__)

    def get(self, index):
        return self.data[index]

    def push(self, item):
        self.data[self.length] = item
        self.length += 1

    def pop(self):
        lastitem = self.data[self.length - 1]
        del self.data[self.length - 1]
        self.length -= 1
        return lastitem

    def delete(self, index):
        deleteditem = self.data[index]
        for i in range(index, self.length - 1):
            self.data[i] = self.data[i + 1]

        del self.data[self.length - 1]
        self.length -= 1
        return deleteditem


arr = Array()
arr.push(1)
arr.push("Hello World")
arr.push(2)
arr.push(3)
arr.push("Hello")
arr.push("World")
arr.delete(3)
print(arr)

File: Arrays/myOwnArray/myOwnArray.js
class MyArray {
  constructor() {
    this.length = 0;
    this.data = {};
  }

  get(index) {
    return this.data[index];
  }

  push(item) {
    this.data[this.length] = item;
    this.length++;
    return this.length;
  }
  pop() {
    const item = this.data[this.length - 1];
    delete this.data[this.length - 1];
    this.length--;
    return item;
  }
  deleteAtIndex(index) {
    const item = this.data[index];
    this.shiftItemsLeft(index);
    return item;
  }
  shiftItemsLeft(index) {
    for (let i = index; i < this.length - 1; i++) {
      this.data[i] = this.data[i + 1];
    }
    console.log(this.data[this.length - 1]);
    delete this.data[this.length - 1];
    this.length--;
  }
}

const ricky = new MyArray();

function never(array) {
  array.push("never");
  array.push("gonna");
  array.push("!");
}
function letDown(array) {
  array.push("let");
  array.push("you");
  array.push("down");
  array.push("!");
}
function giveUp(array) {
  array.push("give");
  array.push("you");
  array.push("up");
  array.push("!");
}

never(ricky);
ricky.pop();
giveUp(ricky);
ricky.pop();
ricky.push("always");
never(ricky);
ricky.pop();
letDown(ricky);
ricky.deleteAtIndex(5);

console.log(ricky);
